File: MaxSubarraySum.c
#include <stdio.h>
int maxSubarraySum(int arr[], int n) {
    int max_sum = arr[0];  // Tổng dãy con lớn nhất
    int current_sum = arr[0];  // Tổng dãy con kết thúc tại phần tử hiện tại
    for (int i = 1; i < n; i++) {
        // Tính tổng dãy con kết thúc tại phần tử i
        current_sum = (current_sum + arr[i] > arr[i]) ? (current_sum + arr[i]) : arr[i];
        // Cập nhật tổng dãy con lớn nhất nếu cần
        if (current_sum > max_sum) {
            max_sum = current_sum;
        }
    }
    return max_sum;
}

int main() {
    int n; printf("Nhập số lượng phần tử mảng: "); scanf("%d", &n);
    int arr[n]; printf("Nhập các phần tử mảng: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Gọi hàm tính tổng dãy con lớn nhất và in kết quả
    printf("Tổng con lớn nhất trong mảng là: %d\n", maxSubarraySum(arr, n));

    return 0;
}

File: MaxSubarraySum.py
def maxSubarraySum(arr):
    # Gán giá trị khởi tạo cho tổng con lớn nhất
    max_sum = arr[0]
    current_sum = arr[0]

    for i in range(1, len(arr)):
        # Tính tổng con lớn nhất kết thúc tại chỉ số i
        current_sum = max(current_sum + arr[i], arr[i])
        
        # Cập nhật tổng con lớn nhất toàn cục
        max_sum = max(max_sum, current_sum)
    
    return max_sum

# Nhập mảng từ người dùng
arr = list(map(int, input("Nhập dãy số (cách nhau bởi dấu cách): ").split()))
print("Tổng dãy con lớn nhất là:", maxSubarraySum(arr))
